### PR TITLE
Add Host-scanner onboarding script

### PR DESCRIPTION
**Pull request by frasecci**

### CODE REVIEW

File: modules/services/host-scanner/main.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+provider "azurerm" {
```
Comment (mdkulkarni15) [inline modules/services/host-scanner/main.tf:1]: won't you need any output.tf?

File: modules/services/host-scanner/main.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  subscription_id = var.subscription_id
+}
+
+resource "azurerm_lighthouse_definition" "lighthouse_definition" {
+  name               = "Sysdig Agentless Host Scanner"
+  description        = "Lighthouse definition for Sysdig Agentless Host Scanner"
+  managing_tenant_id = var.sysdig_tenant_id
+  scope              = "/subscriptions/${var.subscription_id}"
+  authorization {
+    principal_id           = var.sysdig_service_principal_id
```
Comment (mdkulkarni15) [inline modules/services/host-scanner/main.tf:15]: who is creating this service principal? customer?

File: modules/services/host-scanner/versions.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  required_version = ">= 1.0.0"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.76.0"
+    }
+    sysdig = {
+      source  = "sysdiglabs/sysdig"
+      version = ">= 1.15.0"
```
Comment (mdkulkarni15) [inline modules/services/host-scanner/versions.tf:11]: you need to bump this to 1.18.2 i think.

File: modules/services/host-scanner/versions.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 1.0.0"
+
+  required_providers {
+    azurerm = {
```
Comment (mdkulkarni15) [inline modules/services/host-scanner/versions.tf:5]: don't you need azuread as well?

File: modules/services/host-scanner/main.tf [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+data "azurerm_subscription" "primary" {
+  subscription_id = var.subscription_id
+}
+
+resource "azurerm_lighthouse_definition" "lighthouse_definition" {
+  name               = "Sysdig Agentless Host Scanner"
+  description        = "Lighthouse definition for Sysdig Agentless Host Scanner"
+  managing_tenant_id = var.sysdig_tenant_id
```
Comment (mdkulkarni15) [inline modules/services/host-scanner/main.tf:12]: are you planning to use management groups at some point? maybe you can use management groups here too?

File: modules/services/host-scanner/main.tf (hunk 2 above)
Comment (frasecci) [reply modules/services/host-scanner/main.tf:15]: The service principal is the sysdig one, we are not creating any SP on the customer side

File: modules/services/host-scanner/main.tf (hunk 1 above)
Comment (frasecci) [reply modules/services/host-scanner/main.tf:1]: I added it to display the ID of the created resources

File: modules/services/host-scanner/main.tf (hunk 5 above)
Comment (frasecci) [reply modules/services/host-scanner/main.tf:11]: I created an Epic for it https://sysdig.atlassian.net/browse/SSPROD-34668 , for CA we will proceed without management groups

**Review by mdkulkarni15 — APPROVED** [review]

**Pull request merged**